### ISSUE TITLE
Supporting .gitignore

### ISSUE THREAD

**Issue by timmmmyboy** [woolfg/dokuwiki-plugin-gitbacked#23]
Currently if we have private directories in our wiki and we add them to .gitignore we get the following error when attempting to save to pages in that directory:
```

Fatal error: Uncaught exception 'Exception' with message 'The following paths are ignored by one of your .gitignore files: dkc-private.txt Use -f if you really want to add them.
fatal: no files added ' in /main/dtlt/public_html/docs/lib/plugins/gitbacked/lib/Git.php:218
Stack trace: #0 /main/dtlt/public_html/docs/lib/plugins/gitbacked/lib/Git.php(233): GitRepo->run_command('/usr/bin/git --...')
#1 /main/dtlt/public_html/docs/lib/plugins/gitbacked/lib/Git.php(247): GitRepo->run('add /main/dtlt/...')
#2 /main/dtlt/public_html/docs/lib/plugins/gitbacked/action/editcommit.php(86): GitRepo->add('/main/dtlt/publ...')
#3 /main/dtlt/public_html/docs/lib/plugins/gitbacked/action/editcommit.php(172): action_plugin_gitbacked_editcommit->commitFile('/main/dtlt/publ...', 'Edited dkc-priv...')
#4 /main/dtlt/public_html/docs/inc/events.php(204): action_plugin_gitbacked_editcommit->handle_io_wikipage_write(Object(Doku_Event), NULL)
#5 /main/dtlt/public_html/docs/inc/events.php(76): Doku_Event_Handler->process_event(Object(Doku_Event), 'A in /main/dtlt/public_html/docs/lib/plugins/gitbacked/lib/Git.php on line 218
```

Any ideas what the best way to approach private pages in a wiki would be in terms of interfacing with a public repo?

**Comment by woolfg** [comment]
I added a new settings field to specify ignorePaths - see branch https://github.com/woolfg/dokuwiki-plugin-gitbacked/tree/priv_pages - please can you chek, if this enhancement works for your use case

**Comment by timmmmyboy** [comment]
This worked perfectly! Thank you so much for the amazingly quick response.

**Comment by woolfg** [comment]
thx, merged in #24
